pandas/tests/groupby/test_groupby.py

```diff
@@ -223,13 +223,13 @@ def f(x, q=None, axis=0):
     agg_result = df_grouped.agg(np.percentile, 80, axis=0)
     apply_result = df_grouped.apply(DataFrame.quantile, .8)
     expected = df_grouped.quantile(.8)
-    assert_frame_equal(apply_result, expected)
+    assert_frame_equal(apply_result, expected, check_names=False)
```

Comment:
There is some inconsistency between the `apply_result` and the standard / agg counterparts, whereby the former is adding `.8` as the name. I don't think that's necessarily desirable but orthogonal to this change as the root cause there is somewhere within `.apply`

Reply:
hmm, no this is definitily an api change you are making here, and not a good one. you *do* need to set the group name on the resulting index. we do this on purpose.

Reply:
What is the purpose of the API to set the index name to the value of the `q` argument via `apply` but not `agg`? 

Note that the `check_names=False` is applied to `agg_result` but I would think if anything the expectation is backwards on these, as I'm not aware of any other function that would do this